ask-sdk-runtime/src/com/amazon/ask/request/dispatcher/impl/BaseRequestDispatcher.java

```diff
@@ -145,8 +145,8 @@ private Output doDispatch(final Input input) throws Exception {
             LOGGER.debug("Found matching handler");
         } else {
             String message = "Unable to find a suitable request handler";
-            LOGGER.error(message);
-            throw new AskSdkException(message);
+            LOGGER.debug(message);
+            return null;
```

Comment:
By changing this behavior, how does it affect the case where the request handler is not present? Does the service return a response for that or just fail? How will the skill developer know the issue?

Reply:
So we will now rely on this exception to be thrown - https://github.com/alexa/alexa-skills-kit-sdk-for-java/blob/3424376c148b88a5a7b82311c995151154f879b9/ask-sdk-lambda-support/src/com/amazon/ask/SkillStreamHandler.java#L92

This will be thrown once we have iterated over the collection of skills.

Reply:
This is I think what it should have been if this worked as expected.

Reply:
> So we will now rely on this exception to be thrown -
> 
> https://github.com/alexa/alexa-skills-kit-sdk-for-java/blob/3424376c148b88a5a7b82311c995151154f879b9/ask-sdk-lambda-support/src/com/amazon/ask/SkillStreamHandler.java#L92
> 
> This will be thrown once we have iterated over the collection of skills.

That seems to be for `lambda`. How about for the skill servlet package? Is it handled similarly in that workflow as well?

Reply:
Thats a great point! I see that the servlet is not taking a skill collection https://github.com/alexa/alexa-skills-kit-sdk-for-java/blob/3424376c148b88a5a7b82311c995151154f879b9/ask-sdk-servlet-support/src/com/amazon/ask/servlet/SkillServlet.java#L79